index.ts

```diff
@@ -0,0 +1,217 @@
+            const sanitizedParameters: adapter.Parameters = removeKeys(
+              templateParameters || {},
+              RESERVED_PARAMETERS
+            );
+
+            const options: adapter.BuildRequestOptions = {
+              endpointName,
+              parameters: { ...sanitizedParameters, from: "ETH" }, // TODO: fix hardcoded from param
+              metadataParameters: {}, // TODO: fix hardcoded values
+              ois: oisByTitle,
+              apiCredentials: adapterApiCredentials,
+            };
+
+            const apiResponse = await adapter.buildAndExecuteRequest(options);
+            if (!apiResponse || !apiResponse.data) {
+              console.log("Error: failed to fetch data from API");
+              return;
+            }
+            console.log("Info: API server value", apiResponse.data);
+            if (apiResponse.data === 0) {
+              console.log("Error: API responded with value of 0");
+              return;
+            }
+
+            let apiValue: ethers.BigNumber;
+            try {
+              const response = adapter.extractAndEncodeResponse(
+                apiResponse.data,
+                reservedParameters as adapter.ReservedParameters
+              );
+              apiValue = ethers.BigNumber.from(
+                adapter.bigNumberToString(response.value as any)
+              );
+            } catch (e) {
+              console.log("Error: failed to extract data from API response");
+              return;
+            }
+
+            // **************************************************************************
+            // 4. Read beacon
+            // **************************************************************************
+            const beaconResponse = await rrpBeaconServer
+              .connect(airnodeWallet)
+              .readBeacon(templateId);
+            // const beaconResponse = { value: ethers.BigNumber.from("683392028") };
+
+            if (!beaconResponse) {
+              console.log("Error: failed to fetch data from beacon server");
+              return;
+            }
+            console.log("Info: beacon server value", beaconResponse.value);
+
+            // **************************************************************************
+            // 5. Check deviation
+            // **************************************************************************
+            const delta = beaconResponse.value.sub(apiValue).abs();
+            if (delta.eq(0)) {
+              console.log("Info: beacon is up-to-date. skipping update");
+              return;
+            }
+
+            const deviation = delta
+              .mul(100 * Number(reservedParameters._times)) // TODO: can _times be null or 0?
+              .div(apiValue);
+            console.log(
+              "Info: deviation %",
+              deviation.toNumber() / Number(reservedParameters._times)
+            );
```

Comment:
I wonder if we should do this with a static call to https://github.com/api3dao/airnode/blob/psp-draft-v4/packages/airnode-protocol/contracts/psp/BeaconServer.sol#L326
Probably not though

Reply:
That's actually an interesting idea. I forgot about that BeaconServer and it's architecture. We would need to create a subscription per keeper trigger I'm thinking 🤔